Apps/Admin/Client/Store/Analytics/AnalyticsActions.cs

```diff
@@ -0,0 +1,155 @@
+        /// <param name="endDate">Optional end date to include in the query.</param>
+        public LoadNotesAction(DateTime? startDate, DateTime? endDate)
+            : base(startDate, endDate)
+        {
+        }
+    }
+
+    /// <summary>
+    /// The action representing the initiation of ratings load.
+    /// </summary>
+    public class LoadRatingsAction : AnalyticsBaseAction
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="LoadRatingsAction"/> class.
+        /// </summary>
+        /// <param name="startDate">Optional start date to include in the query.</param>
+        /// <param name="endDate">Optional end date to include in the query.</param>
+        public LoadRatingsAction(DateTime? startDate, DateTime? endDate)
+            : base(startDate, endDate)
+        {
+        }
+    }
+
+    /// <summary>
+    /// The action representing the initiation of inactive users load.
+    /// </summary>
+    public class LoadInactiveUsersAction
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="LoadInactiveUsersAction"/> class.
+        /// </summary>
+        /// <param name="inactiveDays">The days inactive to filter the users last login.</param>
+        /// <param name="timeOffset">The offset from the client browser to UTC.</param>
+        public LoadInactiveUsersAction(int inactiveDays, int timeOffset)
+        {
+            this.InactiveDays = inactiveDays;
+            this.TimeOffset = timeOffset;
+        }
+
+        /// <summary>
+        /// Gets or sets inactive days.
+        /// </summary>
+        public int InactiveDays { get; set; }
+
+        /// <summary>
+        /// Gets or sets time offset.
+        /// </summary>
+        public int TimeOffset { get; set; }
+    }
+
+    /// <summary>
+    /// The action representing a successfull action load.
+    /// </summary>
+    public class LoadSuccessAnalyticsAction : BaseLoadSuccessAction<HttpContent>
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="LoadSuccessAnalyticsAction"/> class.
+        /// </summary>
+        /// <param name="state">analytics state.</param>
+        public LoadSuccessAnalyticsAction(HttpContent state)
+            : base(state)
+        {
+        }
+    }
+
+    /// <summary>
+    /// The action representing a failed action load.
+    /// </summary>
+    public class LoadFailAnalyticsAction : BaseLoadFailAction
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="LoadFailAnalyticsAction"/> class.
+        /// </summary>
+        /// <param name="error">The request error.</param>
+        public LoadFailAnalyticsAction(RequestError error)
+            : base(error)
+        {
+        }
+    }
+
+    /// <summary>
+    /// The action that clears the generic report state.
+    /// </summary>
+    public class ResetAnalyticsStateAction
```

Comment:
Nit: "ResetAnalyticsStateAction" should be "ResetStateAction", since it is contained within "AnalyticsActions".